File: py-backend/firebase_init.py
import firebase_admin
from firebase_admin import credentials, firestore
import os

# Construct the absolute path to the service account key file
# This assumes firebase_init.py is in the same directory as the key file
script_dir = os.path.dirname(os.path.abspath(__file__))
key_file_path = os.path.join(script_dir, "shavzak-firebase-adminsdk.json")

# Initialize Firebase Admin - This will run once when this module is first imported
if not firebase_admin._apps: # Check if already initialized to prevent re-initialization
    cred = credentials.Certificate(key_file_path)
    firebase_admin.initialize_app(cred)

db = firestore.client() # Firestore database client

File: backend/src/firebaseInit.ts
import * as admin from 'firebase-admin';
import path from 'path';

// Construct the absolute path to the service account key file
// This assumes your service account key (e.g., shavzak-firebase-adminsdk.json) 
// is located in the project root: /Users/dvircooper/projects/shavzak/
// Adjust the path if your key is located elsewhere.
const serviceAccountPath = path.resolve(__dirname, '../../shavzak-firebase-adminsdk.json');

// Initialize Firebase Admin
// Check if already initialized to prevent errors, especially during hot-reloading in development
if (!admin.apps.length) {
  const serviceAccount = require(serviceAccountPath); // Using require for JSON is common in Node.js
  admin.initializeApp({
    credential: admin.credential.cert(serviceAccount),
  });
}

export const db = admin.firestore(); // Export the initialized Firestore instance